File: QuickShort.cs
using System;

class Program
{
    // Función para intercambiar valores
    static void Intercambiar(ref int a, ref int b)
    {
        int temp = a;
        a = b;
        b = temp;
    }

    // Función de partición
    static int Particion(int[] arreglo, int inicio, int fin)
    {
        int pivote = arreglo[fin]; // Tomamos el último elemento como pivote
        int i = inicio - 1;

        for (int j = inicio; j < fin; j++)
        {
            if (arreglo[j] < pivote)
            {
                i++;
                Intercambiar(ref arreglo[i], ref arreglo[j]);
            }
        }
        Intercambiar(ref arreglo[i + 1], ref arreglo[fin]);
        return i + 1;
    }

    // Función QuickSort
    static void QuickSort(int[] arreglo, int inicio, int fin)
    {
        if (inicio < fin)
        {
            int indicePivote = Particion(arreglo, inicio, fin);

            // Ordenamos recursivamente los elementos a la izquierda y derecha del pivote
            QuickSort(arreglo, inicio, indicePivote - 1);
            QuickSort(arreglo, indicePivote + 1, fin);
        }
    }

    static void Main()
    {
        Console.Write("¿Cuántos elementos quieres ordenar? ");
        int n = int.Parse(Console.ReadLine());

        int[] arreglo = new int[n];
        Console.WriteLine($"Ingresa los {n} elementos:");

        for (int i = 0; i < n; i++)
        {
            arreglo[i] = int.Parse(Console.ReadLine());
        }

        QuickSort(arreglo, 0, n - 1);

        Console.Write("Arreglo ordenado: ");
        for (int i = 0; i < n; i++)
        {
            Console.Write(arreglo[i] + " ");
        }
        Console.WriteLine();
    }
}

File: QuickShort.js
// Función para intercambiar valores
function intercambiar(arr, i, j) {
    let temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

// Función de partición
function particion(arr, inicio, fin) {
    let pivote = arr[fin]; // Tomamos el último elemento como pivote
    let i = inicio - 1;

    for (let j = inicio; j < fin; j++) {
        if (arr[j] < pivote) {
            i++;
            intercambiar(arr, i, j);
        }
    }

    intercambiar(arr, i + 1, fin);
    return i + 1;
}

// Función QuickSort
function quickSort(arr, inicio, fin) {
    if (inicio < fin) {
        let indicePivote = particion(arr, inicio, fin);

        // Ordenamos recursivamente los elementos a la izquierda y derecha del pivote
        quickSort(arr, inicio, indicePivote - 1);
        quickSort(arr, indicePivote + 1, fin);
    }
}

// Programa principal
function main() {
    let n = parseInt(prompt("¿Cuántos elementos quieres ordenar?"));
    let arreglo = [];

    for (let i = 0; i < n; i++) {
        let num = parseInt(prompt(`Ingresa el elemento ${i + 1}:`));
        arreglo.push(num);
    }

    quickSort(arreglo, 0, n - 1);

    console.log("Arreglo ordenado:", arreglo);
}

// Ejecutar el programa
main();
